ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/api/useRunReport.ts

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useState } from 'react';
+
+import { runReportRequest } from 'services/ReportsService';
+import { ReportNotificationMethod } from 'services/ReportsService.types';
+
+export type UseSaveReportProps = {
+    onCompleted: () => void;
+};
+
+type Result = {
+    isRunCompleted: boolean;
+    isRunning: boolean;
+    runError: string | null;
+};
+
+type SaveReportResult = {
+    runReport: (reportConfigId: string, reportNotificationMethod: ReportNotificationMethod) => void;
+} & Result;
+
+const defaultResult = {
+    isRunCompleted: false,
+    isRunning: false,
+    runError: null,
+};
+
+function useRunReport({ onCompleted }: UseSaveReportProps): SaveReportResult {
+    const [result, setResult] = useState<Result>(defaultResult);
+
+    const runReport = useCallback(
+        (reportConfigId: string, reportNotificationMethod: ReportNotificationMethod) => {
+            setResult({
+                isRunCompleted: false,
+                isRunning: true,
+                runError: null,
+            });
+
+            runReportRequest(reportConfigId, reportNotificationMethod)
+                .then(() => {
+                    setResult({
+                        isRunCompleted: true,
+                        isRunning: false,
+                        runError: null,
+                    });
+                    onCompleted();
+                })
+                .catch((err) => {
+                    setResult({
+                        isRunCompleted: true,
+                        isRunning: false,
+                        runError: err.response.data.message,
+                    });
+                });
```

Comment:
I think `err` here is inferred as type `any`, and it is possible that `err.response` will be undefined throwing a runtime error. You could wrap this as `getAxiosErrorMessage(err)` from `utils/responseErrorsUtils` which would return the same value, if defined, but return some fallbacks in case something else goes wrong before the server returns a response.

Reply:
Unrelated nit: my gut feeling is that if the request failed or returned an error the `isRunCompleted` value would be `false` instead of `true`.